var/spack/repos/builtin/packages/mercurial/package.py

```diff
@@ -47,7 +48,9 @@ class Mercurial(PythonPackage):
     version('3.8.2', 'c38daa0cbe264fc621dc3bb05933b0b3')
     version('3.8.1', '172a8c588adca12308c2aca16608d7f4')
 
-    depends_on('python@2.6:2.8')
+    depends_on('python')
+    depends_on('python@2.6:2.8', when='@:4.2.99')
+    conflicts('python@3.0:3.4')
```

Comment:
This shouldn't be in `conflicts`, it should be in `depends_on`.

Reply:
Done but I don't quite get the idea of conflicts() then. My reasoning was:
- In general we depend on python (i.e. all versions need some form of python) => `depends_on('python')`
- If the version is below 4.3 python 2 is required => `depends_on('python@2.6:2.8')`
- There is no support for python 3 below 3.5 => `conflicts('python@3.0:3.4')`